### PR TITLE
[DDP][Compile] Test to Ensure torch.compile works w/static_graph=True

### PR DESCRIPTION
**Pull request by rohan-varma**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114621

Resolves https://github.com/pytorch/pytorch/issues/93672. This was
actually fixed by https://github.com/pytorch/pytorch/pull/103487 but I didn't
realize that PR also fixes torch compile at the time.

Differential Revision: [D51596148](https://our.internmc.facebook.com/intern/diff/D51596148/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @kiukchung @d4l3k @lucasllc

### CODE REVIEW

File: torch/testing/_internal/distributed/distributed_test.py [hunk 1]
```diff
@@ -9566,6 +9566,38 @@ def abort(device):
...
+            model = torch.nn.Linear(10, 10).cuda(self.rank)
+            model_clone = copy.deepcopy(model)
+            ddp = torch.nn.parallel.DistributedDataParallel(
+                model,
+                device_ids=[self.rank],
+            )
+            ddp_static = torch.nn.parallel.DistributedDataParallel(
+                model_clone,
+                device_ids=[self.rank],
+                static_graph=True
```
[the remark this answers is not in the document]
Comment (davidberard98) [reply torch/testing/_internal/distributed/distributed_test.py:9586]: I think it was for every model (but hf_bert is probably still a good idea?)

and thanks for adding the test!

**Review by wconstab — APPROVED** [review]
lgtm, thanks @rohan-varma !

**Comment by rohan-varma** [comment]
Inductor accuracy failure looks unrelated: https://github.com/pytorch/pytorch/actions/runs/7010447659/job/19072184242.
